File: go/reward.go
package fit

// Reward represents a scoring result (reward-schema-v1).
type Reward struct {
	Score     float64        `json:"score" yaml:"score"`
	Breakdown map[string]float64 `json:"breakdown" yaml:"breakdown"`
	Metadata  map[string]any `json:"metadata,omitempty" yaml:"metadata,omitempty"`
}

// RewardScorer scores frontier LLM output.
type RewardScorer interface {
	Score(output string, context map[string]any) (*Reward, error)
}

File: rs/src/reward.rs
use serde::{Deserialize, Serialize};
use std::collections::HashMap;

/// Reward scoring result (reward-schema-v1).
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Reward {
    pub score: f64,
    pub breakdown: HashMap<String, f64>,
    #[serde(default)]
    pub metadata: HashMap<String, serde_yaml::Value>,
}

/// Reward scorer trait.
pub trait RewardScorer: Send + Sync {
    fn score(
        &self,
        output: &str,
        context: &HashMap<String, serde_yaml::Value>,
    ) -> Result<Reward, Box<dyn std::error::Error + Send + Sync>>;
}
